automation/Extension/firefox/lib/http-post-parser.js

```diff
@@ -0,0 +1,250 @@
+// Incorporates code from: https://github.com/redline13/selenium-jmeter/blob/6966d4b326cd78261e31e6e317076569051cac37/content/library/recorder/HttpPostParser.js
+
+const {Cc, Ci, CC, Cu, components} = require("chrome");
+var loggingDB     = require("./loggingdb.js");
+
+var HttpPostParser = function(stream) {
+  // Scriptable Stream Constants
+  this.seekablestream = stream;
+  this.stream = components.classes["@mozilla.org/scriptableinputstream;1"].createInstance(components.interfaces.nsIScriptableInputStream);
+  this.stream.init(this.seekablestream);
+
+  this.postBody = "";
+  this.postLines = [];
+  this.postHeaders = [];
+  // Check if the stream has headers
+  this.hasheaders = false;
+  this.body = 0;
+  if (this.seekablestream instanceof components.interfaces.nsIMIMEInputStream) {
+    this.seekablestream.QueryInterface(components.interfaces.nsIMIMEInputStream);
+    this.hasheaders = true;
+    this.body = -1;
+  } else if (this.seekablestream instanceof components.interfaces.nsIStringInputStream) {
+    this.seekablestream.QueryInterface(components.interfaces.nsIStringInputStream);
+    this.hasheaders = true;
+    this.body = -1;
+  }
```

Comment:
The upload stream can be a `nsIMIMEInputStream`, and `nsIStringInputStream`, or some third option (i.e. the default, which is what happens when line 18 and line 22 of `http-post-parser.js` are false)? The first two of which have which have headers but the default option which won't? This is confusing, what is the default stream interface type and why doesn’t it have headers? Are this third options headers included in the main HTTP headers?